src/chatClient.cpp

```diff
@@ -320,6 +320,35 @@ bool Client::openDb(const std::string& sid)
                 ok = true;
                 KR_LOG_WARNING("Database version has been updated to %s", gDbSchemaVersionSuffix);
             }
+            else if (cachedVersionSuffix == "10" && (strcmp(gDbSchemaVersionSuffix, "11") == 0))
+            {
+                // Remove USER_ATTR_RSA_PUBKEY attr from cache
+                db.query("delete from userattrs where type = 64");
+
+                // Create temporary table and copy sendkeys content
+                db.query("CREATE TABLE tempkeys(chatid int64 not null, userid int64 not null, keyid int32 not null, key blob not null,ts int not null, UNIQUE(chatid, userid, keyid));");
+                db.query("INSERT INTO tempkeys(chatid, userid, keyid, key, ts) SELECT chatid, userid, keyid, key, ts FROM sendkeys");
+
+                // Close and re-open db again to avoid SQLITE_LOCKED
+                db.close();
+                if (!db.open(path.c_str(), false))
+                {
```

Comment:
Better to enclose the queries in the bloc if open is successful, and fallback to the standard error's handling routine in case it fails.

Reply:
I've followed the same approach as here: https://github.com/meganz/MEGAchat/blob/feature/legacy-encryption-code-cleanup/src/chatClient.cpp#L169

Reply:
Yes, but you already opened the DB started to migrate the squema. Also, the log message is the same than the previous one (always try to make log messages unique in the code, so it's easier for debugging).